src/test/java/com/alipay/oceanbase/rpc/ObTableClientTest.java

```diff
@@ -2689,4 +2691,571 @@ public void testAggregationWithIllegalColumn() throws Exception {
+            // ob version is invalid
+            Assert.assertTrue(false);
+        } else if (ObGlobal.OB_VERSION == 4) {
+            // todo: auto increment only support in 4.x currently
+            final ObTableClient client = (ObTableClient) this.client;
+            client.addRowKeyElement("test_auto_increment_rowkey", new String[] { "c1", "c2" });
+
+            // use sql to create table
+            Connection connection = ObTableClientTestUtil.getConnection();
+            Statement statement = connection.createStatement();
+
+            try {
+                statement.execute("CREATE TABLE IF NOT EXISTS test_auto_increment_rowkey_append(c1 int auto_increment,c2 int not null,c3 varchar(255),primary key(c1, c2));");
+
+                // test append not exist, insert
+                Map<String, Object> res = client.append("test_auto_increment_rowkey_append", new Object[] { 0, 1 },
+                        new String[] { "c3" },  new Object[] { "a" } , true);
+
+                TableQuery tableQuery = client.query("test_auto_increment_rowkey_append");
+                tableQuery.select("c1", "c2", "c3");
+                ObTableValueFilter filter = new ObTableValueFilter(ObCompareOp.EQ, "c1", 1);
+                tableQuery.setFilter(filter);
+                QueryResultSet result = tableQuery.execute();
+                result.next();
+                Map<String, Object> value = result.getRow();
+                Assert.assertEquals(1, value.get("c2"));
+                Assert.assertEquals("a", value.get("c3"));
+
+                // test append exist, append
+                res = client.append("test_auto_increment_rowkey_append", new Object[] { 1, 1 },
+                        new String[] { "c3" },  new Object[] { "b" } , true);
+
+                tableQuery = client.query("test_auto_increment_rowkey_append");
+                tableQuery.select("c1", "c2", "c3");
+                filter = new ObTableValueFilter(ObCompareOp.EQ, "c1", 1);
+                tableQuery.setFilter(filter);
+                result = tableQuery.execute();
+                result.next();
+                value = result.getRow();
+                Assert.assertEquals(1, value.get("c2"));
+                Assert.assertEquals("ab", value.get("c3"));
+            } finally { // use sql to drop table
+                statement.execute("drop table test_auto_increment_rowkey_append");
+            }
+        }
+    }
+
+    @Test
+    // Test auto increment on not rowkey
+    public void testAutoIncrementNotRowkey() throws Exception {
+        if (ObGlobal.OB_VERSION <= 0) {
+            // ob version is invalid
+            Assert.assertTrue(false);
+        } else if (ObGlobal.OB_VERSION == 4) {
+            // todo: auto increment only support in 4.x currently
+            final ObTableClient client = (ObTableClient) this.client;
+            client.addRowKeyElement("test_auto_increment_not_rowkey", new String[] { "c1" });
```

Comment:
Make _test_auto_increment_not_rowkey_ a final variable.

Reply:
I do it in prev